src/main.rs

```diff
@@ -228,7 +222,30 @@ fn notmain() -> Result<i32, anyhow::Error> {
         }
     }
 
-    let (range_names, rtt_addr, uses_heap) = range_names_from(&elf, text.index())?;
+    let text = elf
+        .section_by_name(".text")
+        .map(|section| section.index())
+        .ok_or_else(|| {
+            anyhow!(
+            "`.text` section is missing, please make sure that the linker script was passed to the \
+            linker (check `.cargo/config.toml` and the `RUSTFLAGS` variable)"
+        )
+        })?;
```

Comment:
Hmm, if this is done after `elf2table::parse` we might not reach this if the linker script is missing, I think.

Reply:
good point! I'll move it back to its original place.